crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -1036,57 +1024,30 @@ impl RoomInfo {
         Some(&self.base_info.topic.as_ref()?.as_original()?.content.topic)
     }
 
-    /// Get a list of all the valid (non expired) matrixRTC memberships and
-    /// associated UserId's in this room.
-    ///
-    /// The vector is ordered by oldest membership to newest.
-    fn active_matrix_rtc_memberships(&self) -> Vec<(OwnedUserId, &Membership)> {
-        let mut v = self
+    /// Returns an active (on-going) room call (also known as matrix RTC
+    /// session).
+    pub fn active_room_call(&self) -> Option<RoomCall> {
+        let members: BTreeMap<_, _> = self
             .base_info
             .rtc_member
             .iter()
             .filter_map(|(user_id, ev)| {
-                ev.as_original().map(|ev| {
-                    ev.content
-                        .active_memberships(None)
-                        .into_iter()
-                        .map(move |m| (user_id.clone(), m))
-                })
+                let ev = ev.as_original()?;
+                let active = ev.content.active_memberships(None);
+                let room_call_memberships = active.into_iter().filter(|m| m.is_room_call());
+                Some(room_call_memberships.filter_map(|m| {
+                    let id = CallMemberIdentifier {
+                        user_id: user_id.clone(),
+                        device_id: m.device_id.clone().into(),
+                    };
+                    let data = CallMemberData { member_since: m.created_ts?, valid_for: m.expires };
```

Comment:
Does this `?` return None for the whole RoomCall as soon as we have one member without `created_ts`?

This should never be the case except if one forgets to call: `set_created_ts_if_none` I am not sure if we might want to panic then, instead of silently returning `None` for the whole session?

Reply:
`?` is relevant to the closest context (in this case - a closure).

Yeah, actually, at this point `created_at` must be defined and that was my point of complaint that we're using `Membership` type here, although it's not the original `Membership`, but should be a newtype :) We could solve it by not using the `Membership` at this point and replacing it with a different type.